src/main/java/com/smalaca/messagesender/service/UserCrud.java

```diff
@@ -20,30 +27,61 @@ public UserCrud(InMemoryUserRepository userRepository) {
         this.userRepository = userRepository;
     }
 
-    public boolean createUser(UserDto userDto) {
+    public Response createUser(UserDto userDto) {
         User user = new UserFactory().createFrom(userDto);
 
-        if ((user.getLogin().equals("")) || user.getEmail().equals("") && user.getTwitter().equals("") && user.getSlack().equals(""))
-            return false;
-
         try {
+            isUserValid(user);
             userRepository.add(user);
-            return true;
+            return aSuccessfulResponseWith("User Created");
         } catch (UserAlreadyExistException e) {
-            return false;
+            return aFailureResponse("User Creation Failed!");
+        } catch (InvalidUserPassed e) {
+            return aFailureResponse(e.getMessage());
         }
     }
 
-    public boolean blockUser(String login) {
+    public Response blockUser(String login) {
         try {
             userRepository.blockUser(login);
-            return true;
+            return aSuccessfulResponseWith("User Blocked");
         } catch (UserDoesntExistException e) {
-            return false;
+            return aFailureResponse(e.getMessage());
         }
     }
 
     public boolean isUserBlocked(String login) {
         return userRepository.getUserByLogin(login).isBlocked();
     }
+
+    private void isUserValid(User user) {
+        if ((user.getLogin().equals("")) || user.getEmail().equals("") && user.getTwitter().equals("") && user.getSlack().equals(""))
+            throw new InvalidUserPassed();
+    }
+
+    public List<User> showAllUsers() {
+        return userRepository.showAllUsers();
+    }
+
+    public Response updateUser(UserDto userDto) {
+        User user;
+
+        try {
+            user = userRepository.getUserByLogin(userDto.getLogin());
+        } catch (UserDoesntExistException e) {
+            return aFailureResponse(e.getMessage() + " ,can't update.");
+        }
+
+        if (userRepository.updateUser(user, userDto))
+            return aSuccessfulResponseWith("User " + user.getLogin() + " updated");
+        return aFailureResponse("Unexpected Problem");
+    }
+
+    public User showUser(String login) {
+        try {
+            return userRepository.getUserByLogin(login);
+        } catch (UserDoesntExistException e) {
+            return null;
```

Comment:
jeżeli użytkownik nie istenieje to powinieneś albo:
1) puścić wyjątek i go obsłużyć gdzieś indziej
2) złapać wyjątek i obsłużyć go teraz. 

null nie jest odpowiedzią.